assets/arc-0052/bip32-ed25519.ts

```diff
@@ -0,0 +1,180 @@
+import { createHash, createHmac } from "crypto";
+import {
+    crypto_core_ed25519_add,
+  crypto_scalarmult_ed25519_base_noclamp
+} from "libsodium-wrappers-sumo";
+var BN = require("bn.js");
+
+/**
+ *
+ * Reference of BIP32-Ed25519 Hierarchical Deterministic Keys over a Non-linear Keyspace (https://acrobat.adobe.com/id/urn:aaid:sc:EU:04fe29b0-ea1a-478b-a886-9bb558a5242a)
+ *
+ * @see section V. BIP32-Ed25519: Specification;
+ * 
+ * A) Root keys
+ * 
+ * @param seed - 256 bite seed generated from BIP39 Mnemonic
+ * @returns - Extended root key (kL, kR, c) where kL is the left 32 bytes of the root key, kR is the right 32 bytes of the root key, and c is the chain code. Total 96 bytes
+ */
+export function fromSeed(seed: Buffer): Uint8Array {
+  // k = H512(seed)
+  let k: Buffer = createHash("sha512").update(seed).digest();
+  let kL: Buffer = k.subarray(0, 32);
+  let kR: Buffer = k.subarray(32, 64);
+
+  // While the third highest bit of the last byte of kL is not zero
+  while ((kL[31] & 0b00100000) !== 0) {
+    k = createHmac("sha512", kL).update(kR).digest();
+    kL = k.subarray(0, 32);
+    kR = k.subarray(32, 64);
+  }
+
+  // clamp
+  //Set the bits in kL as follows:
+  // little Endianess
+  kL[0] &= 0b11_11_10_00; // the lowest 3 bits of the first byte of kL are cleared
+  kL[31] &= 0b01_11_11_11; // the highest bit of the last byte is cleared
+  kL[31] |= 0b01_00_00_00; // the second highest bit of the last byte is set
+
+  // chain root code
+  const c: Buffer = createHash("sha256").update(Buffer.concat([new Uint8Array(0x01), seed])).digest();
```

Comment:
This line seems to refer to formula $H_{256}(0x01||k)$ in the referenced paper, which is the concatenation of byte strings `0x01` and $k$.

The line `new Uint8Array(0x01)` does not generate a byte string `[0x01]`, but rather an array of zeros with length `0x01`, i.e., `[0x00]`. 

Therefore, should the above line be changed to `new Uint8Array([0x01])`?

Reply:
Great catch. That's definitely an oversight. Thank you!

Reply:
@k13n, Kevin, what an eye!!!!